### PR TITLE
Origin/moore garrett error controller

### PR DESCRIPTION
**Pull request by gmoore10**
Added functionality to log errors to DB table. DB table was created using migrations.

For the next release, additional functionality to enable/disable presenting errors to an admin and fetching the last X number of errors will be implemented.

### CODE REVIEW

**Comment by Kankur0** [comment]
Everything works as intended for this specific release and passes the style guide tests.

We may want to add the two new commands to the help menu and only show it for Admins, but this may be something we have planned for the next release of this addition.

**Review by Kankur0 — APPROVED** [review]
Works as intended and passes Style Guide.

**Comment by reactivepixel** [comment]
Very clean and well thought through approach. Solid work all around! Actual first try!
First Try

**Comment by Kankur0** [comment]
Great work on this, it's very in depth and helpful when looking into errors. I did however find two small things that we may want to change.

First on lines 109 and 110 of the `errors.js` file there are two util logs but no debug level. I'm assuming this was meant to either be removed or given a debug level. Maybe we should modify these two lines by giving them an appropriate level or removing them if they aren't needed.

The second thing is that doing `!fetchErrors <user>` gives me a "Complete" message but doesn't actually give any of the errors from that user.

**Review by Kankur0 — APPROVED** [review]
Errors properly get added to the database and admins are able to toggle alerts of the events on and off. Also allows admins to fetch all errors or by user with or without a limit. Great work!

File: bot/controllers/errors.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      });
+    return true;
+  }
+
+  // Fetches last X errors logged
+  fetchErrors() {
+    const { message } = this;
+    let params = [];
+
+    if (message.parsed[1] !== undefined) {
```
Comment (reactivepixel) [inline bot/controllers/errors.js:96]: Magic Number

File: bot/botUtils.js [hunk 2]
```diff
@@ -20,13 +20,19 @@ exports.generateCode = (n) => {
 };
 
 // Checks if person is an admin user, use GuildMember object
-exports.isAdmin = (member) => {
-  const adminRoles = [
+exports.isInElevatedRole = (member) => {
```
Comment (reactivepixel) [inline bot/botUtils.js:23]: A more well defined name for these users now, good call in making this change.

**Pull request closed without merging**